### PR TITLE
Integrate associate land form into workflow.

### PR DESCRIPTION
**Pull request by devinleighsmith**
update the draft marker synchronizer to work with the new land/building model structure
replace empty "" valuation data with undefined.
remove dead old parcel detail form components.

### CODE REVIEW

File: frontend/src/features/mapSideBar/SidebarContents/BuildingForm.tsx [hunk 1]
```diff
@@ -149,37 +156,61 @@ interface IBuildingForm {
...
   formikRef,
+  setBuildingToAssociateLand,
 }) => {
   const keycloak = useKeycloakWrapper();
   const dispatch = useDispatch();
+  const { createBuilding, updateBuilding } = useBuildingApi();
   const withNameSpace: Function = React.useCallback(
     (name?: string) => {
-      return [nameSpace ?? '', `${index ?? ''}`, name].filter(x => x).join('.');
+      return [nameSpace ?? '', name].filter(x => x).join('.');
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/BuildingForm.tsx:204]: buildings are not part of parcels anymore.

File: frontend/src/features/mapSideBar/SidebarContents/BuildingForm.tsx [hunk 2]
```diff
@@ -203,18 +234,14 @@ const BuidingForm: React.FC<IBuildingForm> = ({
...
-          apiValues.latitude = apiValues.buildings[0].latitude;
-          apiValues.longitude = apiValues.buildings[0].longitude;
-          apiValues.agencyId = apiValues.buildings[0].agencyId;
-          apiValues.classificationId = apiValues.buildings[0].classificationId;
-
           try {
+            let building: IBuilding;
             if (!values.data.id) {
-              await createParcel(apiValues)(dispatch);
+              building = await createBuilding(apiValues)(dispatch);
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/BuildingForm.tsx:240]: create the building directly.

File: frontend/src/features/mapSideBar/SidebarContents/BuildingForm.tsx [hunk 3]
```diff
@@ -203,18 +234,14 @@ const BuidingForm: React.FC<IBuildingForm> = ({
...
             } else {
-              await updateParcel(apiValues)(dispatch);
+              building = await updateBuilding(apiValues)(dispatch);
             }
+            setBuildingToAssociateLand(building);
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/BuildingForm.tsx:244]: trigger the associated land popup.

File: frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx [hunk 4]
```diff
@@ -241,13 +240,6 @@ const LandForm: React.FC<ILandForm> = (props: ILandForm) => {
     }
 
     let errors = await yupErrors;
-    const { buildings: buildingErrors, tabs, ...parcelErrors } = errors;
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx:244]: we don't need this as we don't have parcel/building tabs anymore on this view.

File: frontend/src/features/properties/hooks/useDraftMarkerSynchronizer.tsx [hunk 5]
```diff
@@ -26,25 +20,17 @@ interface IDraftMarker {
...
     {
       latitude: values.latitude,
       longitude: values.longitude,
       name: values.name?.length ? values.name : 'New Parcel',
-      propertyTypeId: PropertyTypes.DRAFT_PARCEL,
+      propertyTypeId:
+        values.parcelId !== undefined ? PropertyTypes.DRAFT_BUILDING : PropertyTypes.DRAFT_PARCEL,
     },
   ];
-  const buildingMarkers =
```
Comment (devinleighsmith) [inline frontend/src/features/properties/hooks/useDraftMarkerSynchronizer.tsx:38]: no longer need this nested parcel-building structure.

File: frontend/src/components/common/GenericModal.tsx [hunk 6]
```diff
@@ -49,7 +49,7 @@ interface ModalProps {
   /** Optional title to display - no default. */
   title?: string;
   /** Optional message to display - no default. */
-  message?: string;
+  message?: string | React.ReactNode;
```
Comment (Fosol) [inline frontend/src/components/common/GenericModal.tsx:52]: Interesting

**Review by Fosol — APPROVED** [review]

File: frontend/src/components/common/GenericModal.tsx (hunk 6 above)
Comment (devinleighsmith) [reply frontend/src/components/common/GenericModal.tsx:52]: Yeah, this is in keeping with what some of the react-bootstrap components do. For example, tab titles can be either strings or nodes. That allows you to embed things like icons in the title. This makes sense for cases where passing the component as a child doesn't really make sense.

File: frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx [hunk 7]
```diff
@@ -259,6 +305,27 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
       hidePolicy={true}
     >
       {render()}
+      <GenericModal
+        message={
+          <>
+            <FloatCheck size={32}></FloatCheck>
```
Comment (jdtoombs) [inline frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx:311]: This is handy, I think there are some other places we can do something similar. Also, I've always just assumed but never knew 100%. Is `React.Fragment` the same thing as doing `<>`?

**Review by jdtoombs — APPROVED** [review]

File: frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx [hunk 8]
```diff
@@ -24,13 +24,42 @@ import AssociatedLandForm from '../SidebarContents/AssociatedLandForm';
...
+  financials: IFinancialYear[];
+}
+export interface IFormParcel extends IParcel {
+  financials: IFinancialYear[];
+  buildings: IFormBuilding[];
+}
+
+const FloatCheck = styled(FaCheckCircle)`
+  margin: 1em;
+  color: #2e8540;
```
Comment (agalbrai) [inline frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx:48]: could you change this to $completed-color?

File: frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx (hunk 7 above)
Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx:311]: yep, its just newer syntax

File: frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx (hunk 8 above)
Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx:48]: I can't actually (at least as far as I know) use css vars with styled components. If this is something you want to try and figure out I'd love to learn how.

**Pull request merged**